app/controllers/productions_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductionsController < ApplicationController
+    def index
+        @productions = Production.all
+    end
+
+    def show
+        @production = Production.find(params[:id])
+    end
+
+
+    def new 
+        @production = Production.new
+    end
+
+    def create
+        @production = Production.new(production_params)
+        # extract inventory_id from @
+        id = params[:production][:inventory_id]
+        @inventory = Inventory.find(id)
+        # @inventory = Inventory.find(params[:inventory_id])
+        
+        # Subtract the quantity used from the inventory quantity if we have enough
+        if @inventory.quantity >= @production.quantity_used
+          @inventory.quantity -= @production.quantity_used
+        end
+        if @production.save && @inventory.save
+          # redirect_to view_productions_path, notice: 'Production was successfully created.'
+          redirect_to productions_path, notice: 'Production was successfully created.'
+        else
+          render :new
+        end
+      end
```

Comment:
def create
  @production = Production.new(production_params)
  @inventory = Inventory.find_by(id: params.dig(:production, :inventory_id))

  if @inventory.blank?
    render :new, alert: 'Inventory not found.'
    return
  end

  if @inventory.quantity >= @production.quantity_used
    @inventory.quantity -= @production.quantity_used
  else
    render :new, alert: 'Insufficient inventory quantity.'
    return
  end

  if @production.save && @inventory.save
    redirect_to productions_path, notice: 'Production was successfully created.'
  else
    render :new, alert: 'Failed to create production.'
  end
end